python-lib/spell_checker/input_output_params.py

```diff
@@ -0,0 +1,110 @@
+import dataiku
+from dataiku.customrecipe import *
+import re
+
+
+def get_input_output():
+    input_dataset = get_input_names_for_role("input_dataset")[0]
+    not_to_be_corrected_dataset = get_input_names_for_role("not_to_be_corrected_dataset")
+
+    not_to_be_corrected_dataset = NotToBeCorrectedDataset(not_to_be_corrected_dataset)
+
+    if not_to_be_corrected_dataset.is_present:
+        not_to_be_corrected_dataset.check()
+
+    output_dataset = get_output_names_for_role("output_dataset")[0]
+
+    return input_dataset, not_to_be_corrected_dataset, output_dataset
+
+
+def get_spell_checker_params(recipe_config, not_to_be_corrected_dataset):
+    def _p(param_name, default=None):
+        return recipe_config.get(param_name, default)
+
+    params = {}
+
+    # General parameters
+    params["text_col_list"] = _p("text_col_list")
+    params["suffix"] = _p("suffix")
+
+    # Language parameters
+    params["language"] = _p("language")
+    params["detect_language"] = _p("detect_language")
+    params["single_language_per_column"] = _p("single_language_per_column")
+    params["constrain_languages"] = _p("constrain_languages")
+
+    if params["detect_language"]:
+        params["single_language_per_column"] = _p("single_language_per_column")
+        params["constrain_languages"] = _p("constrain_languages")
+
+    # Get set of untouched words
+    if not_to_be_corrected_dataset.is_present:
+        params["set_untouched_words"] = list(not_to_be_corrected_dataset.get_df().iloc[:, 0])
```

Comment:
This probably needs a preliminary check that the dataset has rows, and a single column. Else, raise an Exception with a custom error mesage.

Oh, just saw that you take care of that in `NotToBeCorrectedDataset`. Good.